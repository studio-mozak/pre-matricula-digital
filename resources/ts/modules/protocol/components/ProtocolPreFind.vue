<template>
  <div class="col-md-8 mt-5 offset-md-2">
    <h2 class="font-muli-20-primary mb-4">
      Identifique o(a) aluno(a) para encontrar o(s) protocolo(s):
    </h2>
    <p class="text-justify font-hind">
      Selecione uma das opções abaixo para validar se o(a)
      <strong>aluno(a)</strong> já possui alguma inscrição de pré-matrícula.
      Caso não seja possível identificá-lo com a opção selecionada, você pode
      escolher outro campo para validar a existência de protocolo(s).
    </p>
    <div class="mt-5">
      <button
        data-test="btn-1"
        type="button"
        class="btn btn-block btn-outline-primary mb-2"
        @click="navigate('/busca-protocolo-cpf')"
      >
        <span
          class="text-uppercase font-weight-lighter font-hind"
          style="font-size: 16px"
        >
          CPF do(a) aluno(a)
        </span>
      </button>
      <button
        data-test="btn-2"
        type="button"
        class="btn btn-block btn-outline-primary mb-2"
        @click="navigate('/busca-protocolo-rg')"
      >
        <span
          class="text-uppercase font-weight-lighter font-hind"
          style="font-size: 16px"
        >
          RG do(a) aluno(a)
        </span>
      </button>
      <button
        data-test="btn-3"
        type="button"
        class="btn btn-block btn-outline-primary mb-2"
        @click="navigate('/busca-protocolo-certidao')"
      >
        <span
          class="text-uppercase font-weight-lighter font-hind"
          style="font-size: 16px"
        >
          Certidão de nascimento do(a) aluno(a)
        </span>
      </button>
      <button
        data-test="btn-4"
        type="button"
        class="btn btn-block btn-outline-primary mb-2"
        @click="navigate('/busca-protocolo-nome-e-nascimento')"
      >
        <span
          class="text-uppercase font-weight-lighter font-hind"
          style="font-size: 16px"
        >
          Nome e data de nascimento do(a) aluno(a)
        </span>
      </button>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useRouter } from 'vue-router';

const router = useRouter();

const navigate = (path: string) => {
  router.push(path);
};
</script>
