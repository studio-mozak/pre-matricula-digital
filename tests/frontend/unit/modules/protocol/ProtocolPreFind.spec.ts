import { VueWrapper, mount } from '@vue/test-utils';
import ProtocolPreFind from '@/modules/protocol/components/ProtocolPreFind.vue';
import { expect, vi } from 'vitest';

vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));

const spyPush = vi.fn();

const onMounted = async () => {
  const wrapper = mount(ProtocolPreFind, {
    global: {
      plugins: [],
      stubs: [],
    },
  });

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('ProtocolPreFind', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProtocolPreFind>>;
  beforeEach(async () => {
    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('check if exists 5 buttons', () => {
    const buttons = wrapper.findAll('button');

    expect(buttons.length).toBe(4);
  });

  test('when I click in "CPF do(a) aluno(a)" button, it should go to "/busca-protocolo-cpf"', async () => {
    const btn = wrapper.find('[data-test="btn-1"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    expect(spyPush).toHaveBeenCalledTimes(1);
    expect(spyPush).toHaveBeenCalledWith('/busca-protocolo-cpf');
  });

  test('when I click in "RG do(a) aluno(a)" button, it should go to "/busca-protocolo-rg"', async () => {
    const btn = wrapper.find('[data-test="btn-2"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    expect(spyPush).toHaveBeenCalledTimes(1);
    expect(spyPush).toHaveBeenCalledWith('/busca-protocolo-rg');
  });

  test('when I click in "CertidÃ£o de nascimento do(a) aluno(a)" button, it should go to "/busca-protocolo-certidao"', async () => {
    const btn = wrapper.find('[data-test="btn-3"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    expect(spyPush).toHaveBeenCalledTimes(1);
    expect(spyPush).toHaveBeenCalledWith('/busca-protocolo-certidao');
  });

  test('when I click in "Nome e data de nascimento do(a) aluno(a)" button, it should go to "/busca-protocolo-nome-e-nascimento"', async () => {
    const btn = wrapper.find('[data-test="btn-4"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    expect(spyPush).toHaveBeenCalledTimes(1);
    expect(spyPush).toHaveBeenCalledWith('/busca-protocolo-nome-e-nascimento');
  });
});
