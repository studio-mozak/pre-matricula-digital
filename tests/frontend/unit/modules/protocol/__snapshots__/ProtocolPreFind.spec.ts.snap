// Vitest Snapshot v1

exports[`ProtocolPreFind > should match snapshot 1`] = `
<div
  class="col-md-8 mt-5 offset-md-2"
>
  <h2
    class="font-muli-20-primary mb-4"
  >
     Identifique o(a) aluno(a) para encontrar o(s) protocolo(s): 
  </h2>
  <p
    class="text-justify font-hind"
  >
     Selecione uma das opções abaixo para validar se o(a) 
    <strong>
      aluno(a)
    </strong>
     já possui alguma inscrição de pré-matrícula. Caso não seja possível identificá-lo com a opção selecionada, você pode escolher outro campo para validar a existência de protocolo(s). 
  </p>
  <div
    class="mt-5"
  >
    <button
      class="btn btn-block btn-outline-primary mb-2"
      data-test="btn-1"
      type="button"
    >
      <span
        class="text-uppercase font-weight-lighter font-hind"
        style="font-size: 16px;"
      >
         CPF do(a) aluno(a) 
      </span>
    </button>
    <button
      class="btn btn-block btn-outline-primary mb-2"
      data-test="btn-2"
      type="button"
    >
      <span
        class="text-uppercase font-weight-lighter font-hind"
        style="font-size: 16px;"
      >
         RG do(a) aluno(a) 
      </span>
    </button>
    <button
      class="btn btn-block btn-outline-primary mb-2"
      data-test="btn-3"
      type="button"
    >
      <span
        class="text-uppercase font-weight-lighter font-hind"
        style="font-size: 16px;"
      >
         Certidão de nascimento do(a) aluno(a) 
      </span>
    </button>
    <button
      class="btn btn-block btn-outline-primary mb-2"
      data-test="btn-4"
      type="button"
    >
      <span
        class="text-uppercase font-weight-lighter font-hind"
        style="font-size: 16px;"
      >
         Nome e data de nascimento do(a) aluno(a) 
      </span>
    </button>
  </div>
</div>
`;
